lib/puppet/http/service/compiler.rb

```diff
@@ -211,4 +211,109 @@ def get_status(name)
 
     deserialize(response, Puppet::Status)
   end
+
+  #
+  # @api private
+  #
+  # Submit a GET request to retrieve a file stored with filebucket
+  #
+  # @param [String] path The request path, formatted by Puppet::FileBucket::Dipper
+  # @param [String] environment Name of the environment we are operating in.
+  #   This should not impact filebucket at all, but is included to be consistent
+  #   with legacy code.
+  # @param [String] bucket_path
+  # @param [String] diff_with a checksum to diff against if we are comparing
+  #   files that are both stored in the bucket
+  # @param [String] list_all
+  # @param [String] fromdate
+  # @param [String] todate
+  #
+  # @return [Puppet::FileBucket::File] The deserialized file returned from the
+  #   server.
+  #
+  def get_filebucket_file(path, environment:, bucket_path: nil, diff_with: nil, list_all: nil, fromdate: nil, todate: nil)
+    headers = add_puppet_headers('Accept' => 'application/octet-stream')
+
+    response = @client.get(
+      with_base_url("/file_bucket_file/#{path}"),
+      headers: headers,
+      params: {
+        environment: environment,
+        bucket_path: bucket_path,
+        diff_with: diff_with,
+        list_all: list_all,
+        fromdate: fromdate,
+        todate: todate
+      }
```

Comment:
If a param value is `nil`, is the param included on the wire, e.g. `diff_with=&list_all=&...`, or is the param omitted? I think we want the former behavior (assuming the old code omitted the params)?

Reply:
It should filter out params with no value, but that'd be a good test to include. I'll add that.

Reply:
done